evap/staff/tests/test_views.py

```diff
@@ -687,56 +709,41 @@ class TestSemesterDeleteView(WebTestStaffMode):
 
     @classmethod
     def setUpTestData(cls):
+        cls.semester = baker.make(Semester)
         cls.manager = make_manager()
+        cls.post_params = {"semester_id": cls.semester.pk}
 
-    def test_failure(self):
-        semester = baker.make(Semester)
-        baker.make(
-            Evaluation,
-            course=baker.make(Course, semester=semester),
-            state=Evaluation.State.IN_EVALUATION,
-            voters=[baker.make(UserProfile)],
-        )
-        self.assertFalse(semester.can_be_deleted_by_manager)
+    @patch.object(Semester, "can_be_deleted_by_manager", False)
+    def test_invalid_deletion(self):
+        self.app.post(self.url, user=self.manager, params=self.post_params, status=400)
+        self.assertTrue(Semester.objects.filter(pk=self.semester.pk).exists())
 
-        self.app.post(self.url, params={"semester_id": semester.pk}, user=self.manager, status=400)
-        self.assertTrue(Semester.objects.filter(pk=semester.pk).exists())
+    @patch.object(Semester, "can_be_deleted_by_manager", True)
+    def test_valid_deletion(self):
+        self.app.post(self.url, user=self.manager, params=self.post_params)
+        self.assertFalse(Semester.objects.filter(pk=self.semester.pk).exists())
```

Comment:
Foreshadowing my opinion on your base-class suggestion: Once we have that, we could also make `self.assertDeleted` and `self.assertNotDeleted` helpers to de-noise assertions like this one